optuna/study/study.py

```diff
@@ -1111,27 +1111,30 @@ def _log_completed_trial(self, trial: FrozenTrial) -> None:
             else:
                 trial_values = {name: value for name, value in zip(metric_names, trial.values)}
             _logger.info(
-                "Trial {} finished with values: {} and parameters: {}. ".format(
+                "Trial {} finished with values: {} and parameters: {}.".format(
                     trial.number, trial_values, trial.params
                 )
             )
         elif len(trial.values) == 1:
-            best_trial = self.best_trial
+            try:
+                best_trial = self.best_trial
+            except ValueError:
+                # If no feasible trials are completed yet, study.best_trial raises ValueError.
+                best_trial = None
+
             trial_value: float | dict[str, float]
             if metric_names is None:
                 trial_value = trial.values[0]
             else:
                 trial_value = {metric_names[0]: trial.values[0]}
-            _logger.info(
-                "Trial {} finished with value: {} and parameters: {}. "
-                "Best is trial {} with value: {}.".format(
-                    trial.number,
-                    trial_value,
-                    trial.params,
-                    best_trial.number,
-                    best_trial.value,
-                )
+
+            message = (
+                f"Trial {trial.number} finished with value: {trial_value} and parameters: "
+                f"{trial.params}."
             )
+            if best_trial is not None:
+                message += f" Best is trial {best_trial.number} with value: {best_trial.value}."
```

Comment:
A nit: I would say we can simplify the code a little as follows:

```suggestion
            try:
                best_trial = self.best_trial
                message += f" Best is trial {best_trial.number} with value: {best_trial.value}."
            except ValueError:
                # If no feasible trials are completed yet, study.best_trial raises ValueError.
                pass
```